d3a_api_client/aggregator.py

```diff
@@ -138,12 +141,33 @@ def execute_batch_commands(self):
             self._client_command_buffer.clear()
             return self.dispatcher.wait_for_command_response('batch_commands', transaction_id)
 
+    def get_uuid_from_area_name(self, name):
+        uuids = [area_uuid for area_uuid, area_dict in self.latest_grid_tree_flat.items()
+                 if "area_name" in area_dict and area_dict["area_name"] == name]
+        if len(uuids) == 1:
+            return uuids[0]
+        else:
+            raise ValueError(f"There are multiple areas named {name} in the tree")
+
+    def _buffer_grid_tree(self, message):
+        self.latest_grid_tree = message["grid_tree"]
+        self.latest_grid_tree_flat = flatten_info_dict(self.latest_grid_tree)
```

Comment:
Again a bit subjective, but how about making this method a decorator in order to be a bit prettier and not see its call from the body of _on_trade, _on_tick ... methods?

Reply:
I was trying to implement this as a decorator, but then I reached the limit of my implementation skills, because inside this function `self.latest_grid_tree` needs to be accessed and gave up. Will try it one more time when adapting to the comments here.